packages/frontend/src/components/common/modal/DashboardCreateModal.tsx

```diff
@@ -79,25 +131,93 @@ const DashboardCreateModal: FC<DashboardCreateModalProps> = ({
+                                }
+                                options={spaces?.map((space) => ({
+                                    value: space.uuid,
+                                    label: space.name,
+                                }))}
+                            />
+                        </FormGroup>
+                        <Button
+                            minimal
+                            icon="plus"
+                            intent="primary"
+                            onClick={() => setIsCreatingnewSpace(true)}
+                        >
+                            Create new space
+                        </Button>
+                    </>
+                ) : (
+                    <>
+                        <FormGroup
+                            label="Name your space"
+                            labelFor="dashboard-space"
+                        >
+                            <InputGroup
+                                id="dashboard-space"
+                                type="text"
+                                value={newSpaceName}
+                                onChange={(e) =>
+                                    setNewSpaceName(e.target.value)
+                                }
+                                placeholder="eg. KPIs"
+                            />
+                        </FormGroup>
+                        <Button
+                            minimal
+                            icon="arrow-left"
+                            intent="primary"
+                            onClick={() => setIsCreatingnewSpace(false)}
+                        >
+                            Save to existing space
+                        </Button>
+                    </>
+                )}
             </DialogBody>
 
             <DialogFooter
                 actions={
                     <>
-                        <Button onClick={handleClose}>Cancel</Button>
+                        <Button
+                            onClick={(e) => {
+                                handleClose(e);
+                                if (onClose) onClose(e);
+                            }}
+                        >
+                            Cancel
+                        </Button>
 
                         <Button
                             intent={Intent.PRIMARY}
                             text="Create"
                             type="submit"
-                            loading={isCreating}
+                            loading={isCreatingDashboard || isCreatingSpace}
                             onClick={handleConfirm}
-                            disabled={isCreating || !name}
+                            disabled={
+                                isCreatingDashboard ||
+                                isCreatingSpace ||
+                                !dashboardName ||
+                                (isCreatingSpace && !newSpaceName)
```

Comment:
```suggestion
                               !newSpaceName
```
this is sufficient for this line, we assume here that `isCreatingSpace` is `false` 👍